### ISSUE TITLE
Python Bindings: Unsupported Buffer <> type from np.float16

### ISSUE THREAD

**Issue by cassianocasagrande** [halide/Halide#3263]
Following line:
`input_image = hl.Buffer(np.copy(img.astype(np.float16), order='F'))`

Raises:
`ValueError: Unsupported Buffer<> type.`

**Comment by steven-johnson** [comment]
More context is needed for this to make sense. How is `img` declared? Including an entire snippet would be more helpful.

**Comment by cassianocasagrande** [comment]
Following snippet reproduces error:

```
import numpy as np
import halide as hl
img = np.zeros((256, 256, 3), dtype=np.float16, order='F')                      
hl_img = hl.Buffer(img)
```

I checked PyBuffer.cpp and it seems the HANDLE_BUFFER_TYPE(TYPE) on the defined functions misses half/float16 type.

**Comment by steven-johnson** [comment]
(Sorry for delayed response, was out for a couple of weeks)

Ah, that looks like an oversight, plain and simple. It should be simple to add the necessary support to the Python wrapper code.

**Comment by steven-johnson** [comment]
Proposed fix is in https://github.com/halide/Halide/pull/3303 -- could you please try it out at let me know if it addresses your issue?

**Comment by steven-johnson** [comment]
I think this should be fixed by #3303, which has landed; please reopen if there are aspects of this issue that still need work.
